### PR TITLE
Kpi tracker parallelize

### PR DESCRIPTION
**Pull request by abizjak**
## Purpose

Decrease the number of network round trips when inserting transactions.

In particular making a query for ID and then inserting two values, waiting for each of those to complete is inefficient. It takes at least 2 * network latency per transaction to insert a transaction. Multiplied by the number of transactions as it used to be.

The changes are 
- Insert account statistics in a single query without looking up transaction id first.
- Execute all transaction insertions at once to remove the network overhead.

## Checklist

- [ ] My code follows the style of this project.
- [ ] The code compiles without warnings.
- [ ] I have performed a self-review of the changes.
- [ ] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [ ] (If necessary) I have updated the CHANGELOG.

### CODE REVIEW

File: kpi-tracker/src/main.rs [hunk 1]
```diff
@@ -240,53 +250,48 @@ impl PreparedStatements {
...
+
+        let update_contract_stats = client
             .prepare_typed(
-                "INSERT INTO contract_activeness (contract, time) VALUES ($1, date_seconds($2)) \
-                 ON CONFLICT DO NOTHING",
-                &[Type::INT8, Type::INT8],
+                "WITH id_table AS (SELECT contracts.id FROM contracts WHERE index=$1 AND \
+                 subindex=$2 LIMIT 1),
+             ct AS (INSERT INTO contracts_transactions (contract, transaction) VALUES ((SELECT id \
+                 FROM id_table), $3))
```
Comment (soerenbf) [inline kpi-tracker/src/main.rs:273]: Same here

**Review by soerenbf — APPROVED** [review]
Looks nice. We talked about the obsolete overhead of maintaining an index on transaction type for the transactions table. Could we include the removal of this in this PR as well?

**Comment by abizjak** [comment]
> Looks nice. We talked about the obsolete overhead of maintaining an index on transaction type for the transactions table. Could we include the removal of this in this PR as well?

Yes, I have some changes after these that I did not push yet. Sorry. Will request another review once I did.

File: kpi-tracker/src/main.rs [hunk 2]
```diff
@@ -240,53 +250,48 @@ impl PreparedStatements {
...
-            .await?;
-        let insert_account_activeness = client
+        let update_account_stats = client
             .prepare_typed(
-                "INSERT INTO account_activeness (account, time) VALUES ($1, date_seconds($2)) ON \
-                 CONFLICT DO NOTHING",
-                &[Type::INT8, Type::INT8],
+                "WITH id_table AS (SELECT accounts.id FROM accounts WHERE address=$1 LIMIT 1)
+                   , at AS (INSERT INTO accounts_transactions (account, transaction) VALUES \
+                 ((SELECT id FROM id_table), $2))
```
[the remark this answers is not in the document]
Comment (abizjak) [reply kpi-tracker/src/main.rs:260]: It's been some time, but I am not sure what you are referring to here.

Which part of the expression are you referring to?

**Pull request merged**